web/app/Helpers/PageBuilder.php

```diff
@@ -0,0 +1,344 @@
+             */
+            $template = $this->parameterArray['template'];
+
+            /**
+             * If we didn't get a valid value back (not empty, contains only
+             *  letters, numbers, underscores, and slashes), use the Index screen. (nothing)
+             */
+            if ($template == null || preg_match('/^[a-z0-9_\/]+$/i', $template) != 1) {
+                $template = 'Index';
+            }
+
+            /**
+             * We receive the call on the php file as a path using two
+             * underscores instead of a slash.  We need to pass the actual path to
+             * our next set of tasks.
+             */
+            $template = str_replace('__', '/', $template);
+            $this->screenTemplate = $template;
+
+            $this->findModuleClass($this->screenTemplate);
+        }
+    }
+
+    /**
+     * Parse and return the corresponding path for a template or screen.
+     * @param $template
+     * @return void
+     */
+    public function findModuleClass($template)
+    {
+        /**  Retrieve the path to the module. */
+        if ($this->moduleTemplate != null) {
+            $classFilename = PathManager::moduleClass($template);
+        }
+        if ($this->screenTemplate != null) {
+            $classFilename = PathManager::screenClass($this->screenTemplate);
+        }
+
+        /** If the file is there, return the path and name. */
+        if (file_exists($classFilename)) {
+            $moduleClassPath = $classFilename;
+            $moduleClassName = array_slice(explode($classFilename, '/'), -1);
+        } else {
+            /** If it's not there, I guess we're going on an adventure. */
+            if ($this->moduleTemplate != null) {
+                $dir = PathManager::moduleClassDir();
+            }
+            if ($this->screenTemplate != null) {
+                $dir = PathManager::screenClassDir();
+            }
+
+            /** We've got an arbitrary number of slashes that may lead to a file. */
+            $module_path = explode('/', $template);
+
+            /** We're going to work our way backwards from the deepest subfolder to the parent to find it. */
+            for ($i = count($module_path) - 1; $i >= 0; $i--) {
+                $path = '';
+                /** For an incrementing value that's the same size as the max path depth... */
+                for ($k = 0; $k < $i; $k++) {
+                    /**
+                     * Build a string with the most specific path to the
+                     * fallback (DefaultModule.php) file.
+                     */
+                    $path .= $module_path[$k] . '/';
+                    if ($this->moduleTemplate != null) {
+                        $path .= 'DefaultModule.php';
+                    }
+                    if ($this->screenTemplate != null) {
+                        $path .= 'DefaultScreen.php';
+                    }
+                }
+                $classFiles[] = $path;
```

Comment:
I personally prefer `array_push()` notation since I think it's more clear / explicit.